Extension/src/LanguageServer/client.ts

```diff
@@ -1551,6 +1551,12 @@ export class DefaultClient implements Client {
                 if (changedSettings["legacyCompilerArgsBehavior"]) {
                     this.configuration.handleConfigurationChange();
                 }
+                if (changedSettings["default.compilerPath"] || changedSettings["default.configurationProvider"]) {
+                    ui.showCompilerStatusIcon(false);
```

Comment:
Can you add a check for the condition, i.e. `changedSettings["default.compilerPath"] !== undefined || changedSettings["default.compileCommands"] !== undefined || changedSettings["default.configurationProvider"] !== undefined`, since the changed setting might be to change it a setting to undefined?